logstash-core-event-java/spec/event_spec.rb

```diff
@@ -135,4 +135,62 @@
       expect(event["tags"]).to eq(["foo"])
     end
   end
+
+
+  # noop logger used to test the injectable logger in Event
+  # this implementation is not complete because only the warn
+  # method is used in Event.
+  module DummyLogger
+    def self.warn(message)
+      # do nothing
+    end
+  end
+
+  context "logger" do
+
+    let(:logger) { double("Logger") }
+    after(:each) {  LogStash::Event.logger = nil }
+
+    # the following 2 specs are using both a real module (DummyLogger)
+    # and a mock. both tests are needed to make sure the implementation
+    # supports both types of objects.
+
+    it "should set logger using a module" do
+      LogStash::Event.logger = DummyLogger
```

Comment:
as written in the comment above the test, this is needed to avoid regression basically. That is because of the fast that this is going through the JRuby Java Ext wrapper which at first did not work with the mock/double. so I had to adjust the code to be more permissive on the type of object it would accept.

maybe the comment is not clear or detailed enough for wht "why"...?